File: backend/src/main/java/com/kratosgado/chatapp/utils/ApiResponse.java
package com.kratosgado.chatapp.utils;

import org.springframework.http.HttpStatus;

public class ApiResponse {
  private String message;
  private Object data;
  private HttpStatus status;

  public ApiResponse(HttpStatus status, String message) {
    this.status = status;
    this.message = message;
  }

  public ApiResponse(HttpStatus status, String message, Object data) {
    this.status = status;
    this.message = message;
    this.data = data;
  }

  public static ApiResponse ok(String message) {
    return new ApiResponse(HttpStatus.OK, message);
  }

  public static ApiResponse okWithData(String message, Object data) {
    return new ApiResponse(HttpStatus.OK, message, data);
  }

  public static ApiResponse created(String message) {
    return new ApiResponse(HttpStatus.CREATED, message);
  }

  public static ApiResponse createdWithData(String message, Object data) {
    return new ApiResponse(HttpStatus.CREATED, message, data);
  }

}

File: backend/src/main/kotlin/com/kratosgado/chatapp/utils/ApiResponse.kt
package com.kratosgado.chatapp.utils

import org.springframework.http.HttpStatus

data class ApiResponse(
    val status: HttpStatus,
    val message: String,
    val data: Any? = null,
) {
    companion object {
        fun ok(message: String) = ApiResponse(HttpStatus.OK, message)

        fun okWithData(
            message: String,
            data: Any?,
        ) = ApiResponse(HttpStatus.OK, message, data)

        fun created(message: String) = ApiResponse(HttpStatus.CREATED, message)

        fun createdWithData(
            message: String,
            data: Any?,
        ) = ApiResponse(HttpStatus.CREATED, message, data)
    }
}
